compiler/rustc_errors/src/lib.rs

```diff
@@ -1304,14 +1304,10 @@ impl DiagCtxtInner {
             && !self.flags.can_emit_warnings
             && !diagnostic.is_force_warn()
         {
-            if diagnostic.has_future_breakage() {
-                (*TRACK_DIAGNOSTIC)(diagnostic, &mut |_| {});
-            }
             return None;
         }
 
         if matches!(diagnostic.level, Expect(_) | Allow) {
-            (*TRACK_DIAGNOSTIC)(diagnostic, &mut |_| {});
             return None;
```

Comment:
Please have a look at the PR that introduced them. I'm wary of removing them just because they have no effect on tests. This may be an incremental issue without tests. Alternatively verify from the source that this won't have an effect.

Reply:
It was #84762, sort of. @cjgillot, do you understand the purpose of these two `(*TRACK_DIAGNOSTIC)(diagnostic, &mut |_| {})` calls? Removing them has no effect on test behaviour.

@oli-obk: I have removed this commit from the PR, all the other commits should be ok. Good to go?

Reply:
I have done this `TRACK_DIAGNOSTIC` change by itself in #120699, I'm now fairly confident it's correct. Details in that PR.